python/lsst/the/monster/measure_uband_slr_colorterm.py

```diff
@@ -0,0 +1,410 @@
+import numpy as np
+from astropy import units
+from smatch import Matcher
+import esutil
+import matplotlib.pyplot as plt
+import warnings
+
+import lsst.sphgeom as sphgeom
+
+from .refcats import GaiaDR3Info, GaiaXPInfo, GaiaXPuInfo, DESInfo, SkyMapperInfo, PS1Info, VSTInfo
+from .splinecolorterms import ColortermSplineFitter, ColortermSpline, MagSplineFitter
+from .utils import read_stars
+from .refcats import RefcatInfo
+
+
+__all__ = [
+    "UbandSLRSplineMeasurer",
+    "read_uband_combined_catalog",
+]
+
+
+class _CombinedInfo(RefcatInfo):
+    FLAG = -1
+
+    def get_flux_field(self, band):
+        if band == "u":
+            return "ref_u_flux"
+        else:
+            return f"{band}_flux"
+
+
+def read_uband_combined_catalog(
+    gaia_ref_info,
+    cat_info_list,
+    uband_ref_info,
+    htm_pixel_list,
+    testing_mode=False,
+):
+    """Read the uband combined catalog for a set of HTM indices.
+
+    This will combine all the catalogs according to proper ranking, and
+    apply all relevant color terms to put them in the intermediate (DES)
+    system. It will also do any corrections to the reference u-band
+    flux that are necessary.
+
+    Parameters
+    ----------
+    gaia_ref_info : `RefcatInfo`
+        Info for the astrometric (Gaia) reference.
+    cat_info_list : `list` [`RefcatInfo`]
+        Reverse priority list of refcat infos to load.
+    uband_ref_info : `RefcatInfo`
+        Info for the u-band reference catalog.
+    htm_pixel_list : `list` [`int`]
+        List of HTM pixel ids to load.
+    testing_mode : `bool`, optional
+        Running in "testing" mode?
+
+    Returns
+    -------
+    gaia_stars_all : `astropy.table.Table`
+        Table of stars, with fluxes filled in.
+    """
+    u_slr_bands = ["g", "r"]
```

Comment:
Do we want this hardcoded? or we could pass the uband_ref_info.get_color_bands("u") result/refcatInfo